scons/site_tools/hosted.py

```diff
@@ -101,5 +101,12 @@ def generate(env, **kw):
 		"-Woverloaded-virtual",
 	]
 
+	if 'CXXFLAGS' in os.environ:
+		flags = "".join(os.environ['CXXFLAGS'])
+		env['CXXFLAGS'].append(flags)
+	if 'LDFLAGS' in os.environ:
+		flags = "".join(os.environ['LDFLAGS'])
+		env['LINKFLAGS'] = flags
+
```

Comment:
you could try `env.Append(ENV = {'PATH' : os.environ['PATH']})` from [line 56 of the same file](https://github.com/salkinium/xpcc/blob/fa6a986aea37b879e9101a1b36f12888970d2dcd/scons/site_tools/hosted.py#L56).

But in general it is what we are already doing, just with the other environment variables.